### ISSUE TITLE
[RPC-JSON] Bad function call exception on consecutive requests

### ISSUE THREAD

**Issue by lenormf** [mawww/kakoune#714]
Hi,

As soon as a command that causes an error is entered on the standard input of a client in the `json` ui mode, all the subsequent requests will cause an error.

There's an `std::bad_function_call` lying around somewhere in the code that I could trigger sending a `resize` request (params: 100, 30), not sure where that comes from though since I can use that request just fine when pasting a pre-crafted packet on `stdin`, but not when the client in owned by another. __EDIT__: it seems that the second call to the `resize` command makes the client crash, here are the requests that did it for me:
```
{"params": [100, 30], "jsonrpc": "2.0", "method": "resize"}
{"params": [1918, 944], "jsonrpc": "2.0", "method": "resize"}
uncaught exception (St17bad_function_call): bad_function_call
```

**Comment by lenormf** [comment]
Added a few more problems I found.

Even though I originally stroke the item from the above list, I'm still not convinced the protocol doesn't need an `id` that will be reflected in the client's replies. I.e. if a UI decides to use the -currently unimplement- `getoption` command in several consecutive requests, and one of them turns out to fail (e.g. using a wrong option name), the UI needs to be able to pinpoint which request failed to decide whether the error can be displayed and ignored or not.
